portfolio/src/main/webapp/script.js

```diff
@@ -28,103 +28,214 @@ function addRandomGreeting() {
+
 //*********************** SEARCH BAR FUNCTIONS ***********************
-/**
-  * Swaps the loader for the content and enables search
-  */
+
 window.onload = function(){
-    swapLoaderForContent();
+    swapLoaderForContent(); // displays page after loaded
+
+    // Next button event listener, always active
+    listenNext = document.getElementById("nextbutton");
+    listenNext.addEventListener("click", searchHandler);
+
+
+    //Runs search on enter
     document.getElementById('searchbar').onkeydown = function(e){
         if(e.keyCode == 13){
-            runSearch(document.getElementById('searchbar')
-                .value.toLocaleLowerCase());
+            // i only want single word alphanumerics to go through
+            retrievedInput = document.getElementById('searchbar').value;
+            const regex = new RegExp(/^[a-z0-9]+$/i);
+            if (regex.test(retrievedInput) === false){
+                shakeSearchBar();
+            } else {
+                searchSetup();
+            }
         }
     };
 }
 
+/**
+ * Handler function for the event listener
+ */
+function searchHandler(){
+    highlightHandler(matchingNodes, retrievedInput, match_index);
+}
+
+/**
+ * Code to shake the search bar and make it glow
+ * Calls a bunch of css transforms
+ */
+function shakeSearchBar(){
+    const el = document.getElementById('searchbar');
+    el.preventDefault;
+    el.classList.remove('searchbarglow')
+    void el.offsetWidth;
+    el.classList.add('searchbarglow');
+}
+
+
 //*********************** SEARCH IMPLEMENTATION ***********************
+
+
 /**
-  * Retreives the HTML from the source
-  */
-function retrieveHtml(){
-    var source = document.body.innerHTML;
-
-    if (source != null){
-        return source.toLocaleLowerCase();    //toLocaleLowerCase() to ignore caps
-    } else {
-        alert("source is null");
+ * Finds all the occurences of a substring in a string
+ */
+function findNeedleInHaystack(small, big) {
+    // no lowercase causes issues with finding the index
+    small = small.toLocaleLowerCase();
+    big = big.toLocaleLowerCase();
+
+    let smallLen = small.length;
+    if (smallLen === 0) return [];
+
+    let startIndex = 0, index, indices = [];
+
+    while ((index = big.indexOf(small, startIndex)) > -1) {
+        indices.push(index);
+        startIndex = index + smallLen;
     }
+
+    return indices;
 }
 
 /**
-  * Finds the closest tag before the provided index in the html text.
-  */
-function getClosestTag(index, source){
-    while (source.charAt(index) != '>'){
-        if (index == 0){
-            alert("index hit 0");
-            break;
+ * Walks the tree and finds matches,
+ * returns index array of matches.
+ * Will optimize to gather straight from tree
+ * rather than push into new arr to save space
+ */
+function htmlWalkerandMatcher(node, input){
+    let all = [];
+
+    for (node= node.firstChild; node; node=node.nextSibling){
+        if (node.nodeType==3){ //if a text node
```

Comment:
Use declarative comments. It helps to declare the intent of your code rather than describe what it does. This helps folks fix bugs in the future. Compare:
```
// true if the node is a text node
vs
// Locates text nodes. Text nodes have no child nodes. 
```